docarray/array/storage/redis/getsetdel.py

```diff
@@ -120,4 +118,6 @@ def _save_offset2ids(self):
         self._update_offset2ids_meta()
 
     def _clear_storage(self):
-        self._client.flushdb()
+        for _id in self._offset2ids.ids:
+            self._del_doc_by_id(_id)
+        self._client.delete(self._offset2id_key)
```

Comment:
my feeling is this could be optimized abit, have you checked `scan_iter` by a specific pattern (prefix/suffix)?